### PR TITLE
Add CVC5ApiOptionException

### PR DESCRIPTION
**Pull request by nafur**
This PR adds the new `CVC5ApiOptionException`. While the driver does not (and can not) do anything special for the two existing api exceptions, it can (and should) properly inform the user about incorrect command line option usage.
The PR also removes the `UnrecognizedOptionException`. It is purely internal now, and immediately catched by the API wrapper. Having a separate exception for this is no longer useful.

The additional catch block in `main.cpp` is only temporary until option parsing has been migrated to the driver and setting the options is done properly via the API.

### CODE REVIEW

File: src/api/cpp/cvc5.h [hunk 1]
```diff
@@ -118,6 +118,31 @@ class CVC5_EXPORT CVC5ApiRecoverableException : public CVC5ApiException
   }
 };
 
+/**
+ * A option-related API exception.
```
Comment (aniemetz) [inline src/api/cpp/cvc5.h:122]:
```suggestion
 * An option-related API exception.
```

File: src/main/main.cpp [hunk 2]
```diff
@@ -51,14 +51,27 @@ int main(int argc, char* argv[]) {
   Options opts;
   try {
     return runCvc5(argc, argv, opts);
-  } catch(OptionException& e) {
+  }
```
Comment (aniemetz) [inline src/main/main.cpp:54]: Can you maybe include this when formatting?

File: src/options/options_public_template.cpp [hunk 3]
```diff
@@ -502,15 +502,14 @@ std::string get(const Options& options, const std::string& name)
 {
   Trace("options") << "Options::getOption(" << name << ")" << std::endl;
   ${getoption_handlers}$
-
-  throw UnrecognizedOptionException(name);
+  throw OptionException("Unrecognized informational or option key or setting: " + name);
```
Comment (aniemetz) [inline src/options/options_public_template.cpp:505]: "informational"?

**Review by aniemetz — CHANGES_REQUESTED** [review]

File: src/options/options_public_template.cpp (hunk 3 above)
Comment (nafur) [reply src/options/options_public_template.cpp:505]: I have no clue what I wanted the message to say there... I'll fix it.

File: src/options/options_public_template.cpp [hunk 4]
```diff
@@ -502,15 +502,14 @@ std::string get(const Options& options, const std::string& name)
...
 }
 
 void setInternal(Options& opts, const std::string& name,
                                 const std::string& optionarg)
                                 {
 ${setoption_handlers}$
-  throw UnrecognizedOptionException(name);
+  throw OptionException("Unrecognized informational or option key or setting: " + name);
```
Comment (aniemetz) [inline src/options/options_public_template.cpp:512]: Ditto :)

**Review by aniemetz — APPROVED** [review]

**Pull request merged**